src/Compilers/Core/Portable/Syntax/SyntaxList`1.cs

```diff
@@ -315,7 +315,7 @@ private static SyntaxList<TNode> CreateList(List<TNode> items)
                 return default(SyntaxList<TNode>);
             }
 
-            var newGreen = GreenNode.CreateList(items.Select(n => n.Green));
+            var newGreen = GreenNode.CreateList(items, n => n.Green);
```

Comment:
```suggestion
            var newGreen = GreenNode.CreateList(items, static n => n.Green);
```

Consider doing this for all the lambdas in this PR to make it very obvious they can't capture and are cacheable.